.github/workflows/release-branch.yaml

```diff
@@ -1,75 +1,136 @@
+          branch="releases/$minor"
+          if git ls-remote --exit-code origin refs/heads/$branch ; then
+            >&2 echo "branch $branch already exists, aborting"
+            exit 1
+          fi
+          echo "branch $branch does not exist"
+          echo "branch=$branch" >> $GITHUB_OUTPUT
+
+  create-branch:
     runs-on: ubuntu-latest
+    needs: cutoff-preconditions
     permissions:
       contents: write
       id-token: write
       repository-projects: read
     steps:
-    - name: Generate token
-      id: generate_token
-      uses: tibdex/github-app-token@v2
-      with:
-        app_id: ${{ secrets.OCMBOT_APP_ID }}
-        private_key: ${{ secrets.OCMBOT_PRIV_KEY }}
-    - name: Checkout
-      uses: actions/checkout@v4
-      with:
-        fetch-depth: 0
-        token: ${{ steps.generate_token.outputs.token }}
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@v2
+        with:
+          app_id: ${{ secrets.OCMBOT_APP_ID }}
+          private_key: ${{ secrets.OCMBOT_PRIV_KEY }}
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: main
+          fetch-depth: 0
+          token: ${{ steps.generate_token.outputs.token }}
 
-    - name: Create Release Branch
-      run: |
-        set -e
-        git config --global user.name github-actions
-        git config --global user.email '${GITHUB_ACTOR}@users.noreply.github.com'
+      - name: Create Release Branch
+        run: |
+          set -e
+          branch=${{ needs.cutoff-preconditions.outputs.branch }}
+          git checkout -b "$branch"
+          git push origin $branch
 
-        tag="${{github.event.inputs.tag}}"
-        if [ -n "$tag" ]; then
-          if ! git ls-remote --tags --exit-code origin "$tag" >/dev/null; then
-            >&2 echo "tag $tag not found"
-            exit 1
-          fi
-          git fetch origin "$tag"
-          git checkout "$tag"
-        else 
-          if [ "${{ github.ref_type }}" != "tag" ]; then
-            >&2 echo "please run workflow on desired tag to create a release branch for or specify a tag as input"
-            exit 1
-          fi
-        
-          tag="${{ github.ref_name }}"
-        fi 
-        
-        if ! [[ "$tag" =~ ^v?[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
-          >&2 echo "no valid non-pre-release tag $tag"
-          exit 1
-        fi
-        if [ "$tag" == "${tag%.0}" ]; then
-          >&2 echo "please use a non-patch tag"
-          exit 1
-        fi
-        if git ls-remote --exit-code origin refs/heads/releases/$tag ; then
-          >&2 echo "branch releases/$tag already exists"
-          exit 1
-        fi
-        echo "creating release branch for $tag"
-        n="releases/$tag"
-        git checkout -b "$n"
-        v="$(go run ./api/version/generate bump-patch)"
-        echo "$v" > VERSION
-        git add VERSION
-        git commit -m "Prepare Development of v$v"
-        git push origin "$n"
```

Comment:
so version correction on release branch happens later, right?

Reply:
since main will be on the correct version during cutoff that is not needed.

Example:

1. Main is on 0.17.0-dev
2. Now we want to prepare the first candidate for 0.17.0 (rc.1), Thus we trigger this cutoff action
3. In this case, the branch now does not need an additional commit because 0.17.0-dev is correct
4. Main then gets bumped via PR to 0.18.0-dev
5. We can now build on Main (as 0.18.0-dev candidates) or on the release branch (as 0.17.0-dev candidates) or trigger the release (on 0.17.0-dev this then leads to 0.17.0-rc1)